uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -64,13 +65,17 @@ define([
         }
       };
       vm.uiOptions = {
-        isAdmin: false,
         isChangeExpanded: false,
+        isManager: false, //when the dialog is opened by manager or admin
+        isOwner: false, //when the dialog is opened by the owner
         multipleDays: true,
+        contact: '',
         showDatePickerFrom: false,
         showDatePickerTo: false,
         userDateFormat: HR_settings.DATE_FORMAT,
+        selectedStatus: undefined,
         showBalance: false,
+        statusLabel: '',
```

Comment:
But what you have here is a property that we could do without, since the all the data that you need to correctly show the status name is already on the `leaveRequest` object itself.

If you don't want to have a "utility" method on the scope (which i agree with), then you could simply have the statuses on the scope accessible by value (see [here](https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-report.js#L230-L235)) for example, which mean you could simply do something like
```html
<span class="col-xs-6">{{::$ctrl.statuses[$ctrl.leaveRequest.status_id].label}}</span>
```
as a result you don't have neither unnecessary property, nor utility method on the scope